src/main/java/ru/yandex/clickhouse/ClickHouseStatementImpl.java

```diff
@@ -828,13 +842,22 @@ public void closeOnCompletion() throws SQLException {
     public boolean isCloseOnCompletion() throws SQLException {
         return closeOnCompletion;
     }
-    
-    private ClickHouseResultSet createResultSet(InputStream is, int bufferSize, String db, String table, boolean usesWithTotals, 
+
+    private ClickHouseResultSet createResultSet(InputStream is, int bufferSize, String db, String table, boolean usesWithTotals,
     		ClickHouseStatement statement, TimeZone timezone, ClickHouseProperties properties) throws IOException {
     	if(isResultSetScrollable) {
     		return new ClickHouseScrollableResultSet(is, bufferSize, db, table, usesWithTotals, statement, timezone, properties);
     	} else {
     		return new ClickHouseResultSet(is, bufferSize, db, table, usesWithTotals, statement, timezone, properties);
     	}
     }
+
+    private Map<ClickHouseQueryParam, String> addQueryIdTo(Map<ClickHouseQueryParam, String> parameters) {
+        if (parameters.containsKey(ClickHouseQueryParam.QUERY_ID))
+            return parameters;
```

Comment:
In this case queryId field should be set with value form parameters to be able to kill a query with user-defined QUERY_ID

Reply:
I have added an implementation for the case when a user set `QUERY_ID` parameter.